webapp/packages/plugin-authentication/src/PluginBootstrap.ts

```diff
@@ -28,7 +28,7 @@ export class PluginBootstrap extends Bootstrap {
     this.menuService.addCreator({
       menus: [TOP_NAV_BAR_SETTINGS_MENU],
       getItems: (context, items) => {
-        if (this.serverConfigResource.enabledAuthProviders.length > 0 && !this.authInfoService.userInfo) {
+        if (this.serverConfigResource.enabledAuthProviders.length > 0 && this.userInfoResource.isAnonymous) {
```

Comment:
`!this.userInfoResource.isAnonymous && !this.userInfoResource.isAuthenticated`

if we use this check multiple times it's better to add third flag to `UserInfoResource` like `isRealUser` or something like this, so we will have three states:

anonymous - authenticated as anonymous user
real user - authenticated as real user with credentials
authenticated - anonymous || real user